language/stdlib/modules/authenticator.move

```diff
@@ -0,0 +1,69 @@
+// Move version of the Libra Transaction authenticator logic. Supports:
+// - creating an authentication policy `T` from public keys a a threshold
+// - converting an authentication policy to an account authentication key
+// - allowing one of the parties in a multisig to rotate their own key
+module Authenticator {
+    use 0x0::Hash;
+    use 0x0::Signature;
+    use 0x0::Transaction;
+    use 0x0::Vector;
+
+    // An authentication policy
+    struct T {
+        public_keys: vector<vector<u8>>,
```

Comment:
This is very interesting! Side idea: this can be a vector of accounts (addresses) instead may be? Then each individual account can rotate its key locally paying gas for it itself. This module will define the logic of when the keys from individual accounts get copied to update the multi-key. This update logic can have rate-limits, or only allow to rotate the multi-key when at least X individual keys have rotated, etc.

Reply:
Great ideas! I really like delaying the update of the auth key; that seems like a nice solution to the Kostas problem (2). We have to think carefully about who has the authority to force the auth key update, but perhaps allowing any of the addresses in the vector to do it (subject to the rate limiting and X individual keys constraint) is ok.

Reply:
+1 for the delayed update idea! I can sleep now 🤐😴